src/EVM/Fetch.hs

```diff
@@ -238,13 +238,13 @@ type Fetcher s = Query s -> IO (EVM s ())
 -- will be pruned anyway.
 checkBranch :: SolverGroup -> Prop -> Prop -> IO BranchCondition
 checkBranch solvers branchcondition pathconditions = do
-  checkSat solvers (assertProps [(branchcondition .&& pathconditions)]) >>= \case
+  checkSat solvers (assertProps [(branchcondition .&& pathconditions)] False) >>= \case
```

Comment:
Why always `False` here?

Reply:
I didn't want to do abstaction-refinement in the middle of interpretation. I was afraid that the solver times out here due to abst-refine and then it's super-hard to debug. Would adding a TODO here be OK? Basically, I am a bit cautious.

Also, currently there is no easy way to pass the abt-refine configuration (yes/no/partial) here. So it may be quite confusing if I did abst-refinement there, even though the user asked for no abst-refinement. I didn't want to make this PR more complicated by finding a way to pass the configuration here. Maybe I'm wrong. What do you think?

Reply:
I guess we would have to add the option to the `PleaseAskSMT` query type? This might be a little awkward. I think I would expect the command line options here to be respected?

Reply:
As discussed, we'll do this as part of a more global resturcturing. Otherwise, even if done in a precise way, it's about 160 LoC change, which is kinda sad.

Reply:
For the moment, I added a default, that can be changed and will change it throughout the codebase. However, that's not configurable from the command line. We'll fix this later.

The 160-line diff can be seen here: 7402256000b3d9234a91ad4ea7d8493860d01f09